src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -343,12 +323,13 @@ protected string GetActiveProfile(IProjectSubscriptionUpdate projectSubscription
         /// <summary>
         /// Creates the intiial set of settings based on the file on disk 
         /// </summary>
-        protected LaunchSettingsData GetLaunchSettings()
+        protected async Task<LaunchSettingsData> GetLaunchSettingsAsync()
         {
+            string fileName = await GetLaunchSettingsFileAsync().ConfigureAwait(false);
             LaunchSettingsData settings;
-            if (FileManager.FileExists(LaunchSettingsFile))
+            if (FileManager.FileExists(fileName))
             {
-                settings = ReadSettingsFileFromDisk();
+                settings = await ReadSettingsFileFromDiskAsync();
```

Comment:
`ReadSettingsFileFromDiskAsync` does the `fileName` computation again. Instead, we can pass the argument here and reduce the overhead